odoo/tools/view_validation.py

```diff
@@ -124,10 +124,12 @@ def get_domain_identifiers(expr):
             _check(len(elem.elts) == 3,
                    f"segments should have 3 elements, found {len(elem.elts)}")
             lhs, operator, rhs = elem.elts
-            _check(isinstance(operator, ast.Str),
+            _check(isinstance(operator, ast.Constant),
```

Comment:
```suggestion
            _check(isinstance(operator, ast.Constant),
            _check(isinstance(operator, ast.Constant) and isinstance(operator.value, str),
```
It's not obvious to me why you split 
`_check(isinstance(operator, ast.Str)`
in two blocks
`_check(isinstance(operator, ast.Constant),`
`_check(isinstance(operator.value, str),`

with the same text repeated for both. Actually there is even a problem in the second text, the substition `{type(operator.value).__name__}` in the fstring  is not correct.
This is easier to see with IDE colors
![image](https://github.com/odoo/odoo/assets/5822488/4983f516-6d2d-4424-93a2-8c2dafbf086f)

Reply:
No good reason, just applied your suggestion :+1:

Reply:
Ho, actually the error message is not exactly the same, one will return the operator type, the other will return the operator **value** type 

Before this change and before 3.8, the error message would be (I think, didn't tested)

1 ->  `operator should be a string, found Num`
x  -> `operator should be a string, found Name`

But since 3.8, the error message would be 
1 -> `operator should be a string, found Constant`
even if string is also a constant. 

I think i tried to reproduce a similar behavior so that 
1  ->  `operator should be a string, found int` 
x  ->  `operator should be a string, found Name`

But this is not the current behavior, so combining the condition is actually closer to the current error message. We could improve it in another pr if needed but your suggestion is better in stable.